llvm/include/llvm/ADT/APInt.h

```diff
@@ -1895,6 +1896,9 @@ class [[nodiscard]] APInt {
   friend struct DenseMapInfo<APInt, void>;
   friend class APSInt;
 
+  // Make DynamicAPInt a friend so it can access BitWidth directly.
+  friend DynamicAPInt;
```

Comment:
```suggestion
  friend class DynamicAPInt;
```
This seems to be the convention. (Technically you don't need the forward-declare if you do that, but I'd still keep it...)